File: puzzles/medium/js/conway-sequence.js
const ORIGINAL = parseInt(readline());
const LINE_NUMBER = parseInt(readline());

let conwaySequence = [ORIGINAL];

for (let i = 1; i < LINE_NUMBER; i++) {
  let tempSequence = [];
  let count = 0;
  let previous = conwaySequence[0];
  for (let j = 0; j < conwaySequence.length; j++) {
    let number = conwaySequence[j];
    if (number === previous) {
      count += 1;
    } else {
      tempSequence.push(count);
      tempSequence.push(previous);
      previous = number;
      count = 1;
    }
  }

  tempSequence.push(count);
  tempSequence.push(previous);
  conwaySequence = tempSequence;
}

print(conwaySequence.join(" "));

File: puzzles/python3/conway_sequence.py
from typing import List


if __name__ == "__main__":
    original: int = int(input())
    line_number: int = int(input())
    conway_sequence: List[int] = [original]

    for line in range(1, line_number):
        temp_sequence: List[int] = []
        count: int = 0
        previous: int = conway_sequence[0]
        for number in conway_sequence:
            if number == previous:
                count += 1
            else:
                temp_sequence.append(count)
                temp_sequence.append(previous)
                previous = number
                count = 1

        temp_sequence.append(count)
        temp_sequence.append(previous)
        conway_sequence = temp_sequence  # update

    print(" ".join(map(str, conway_sequence)))
